samplers/probability/consistent/tracestate.go

```diff
@@ -187,18 +187,20 @@ func parseOTelTraceState(ts string, isSampled bool) (otelTraceState, error) { //
 	return otts, nil
 }
 
-func parseNumber(key string, input string, maximum uint8) (uint8, error) {
+// Here`strconv.ParseInt` is used to parse the input string into an integer. The bit size is specified as 8, which is the bit size of `int8`. The parsed value is then checked against the maximum value, and if it's greater, an error is returned. If the parsed value is within the allowed range, it's returned as an `int8` value.
+
+func parseNumber(key string, input string, maximum int8) (int8, error) {
```

Comment:
This change doesn't make sense. The output of this is going to be used as a uint8 anyway, so we are now throwing away half the usable space. This change will break builds that haven't been run.

Reply:
Thanks for the response! Do you have any suggestions on how to make it , so that I can try to make those changes